x-pack/plugins/session_view/public/components/session_view/index.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState, ComponentType } from 'react';
+import {
+  EuiEmptyPrompt,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiResizableContainer,
+} from '@elastic/eui';
+import { EuiResizableButtonProps } from '@elastic/eui/src/components/resizable_container/resizable_button';
+import { EuiResizablePanelProps } from '@elastic/eui/src/components/resizable_container/resizable_panel';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { SectionLoading } from '../../shared_imports';
+import { ProcessTree } from '../process_tree';
+import { Process, ProcessEvent } from '../../../common/types/process_tree';
+import { SessionViewDetailPanel } from '../session_view_detail_panel';
+import { SessionViewSearchBar } from '../session_view_search_bar';
+import { useStyles } from './styles';
+import { useFetchSessionViewProcessEvents } from './hooks';
+
+interface SessionViewDeps {
+  // the root node of the process tree to render. e.g process.entry.entity_id or process.session_leader.entity_id
+  sessionEntityId: string;
+  height?: number;
+  jumpToEvent?: ProcessEvent;
+}
+
+/**
+ * The main wrapper component for the session view.
+ */
+export const SessionView = ({ sessionEntityId, height, jumpToEvent }: SessionViewDeps) => {
+  const [isDetailOpen, setIsDetailOpen] = useState(false);
+  const [selectedProcess, setSelectedProcess] = useState<Process | null>(null);
+
+  const styles = useStyles({ height });
+
+  const onProcessSelected = (process: Process) => {
+    setSelectedProcess(process);
+  };
+
+  const [searchQuery, setSearchQuery] = useState('');
+  const [searchResults, setSearchResults] = useState<Process[] | null>(null);
+
+  const {
+    data,
+    error,
+    fetchNextPage,
+    hasNextPage,
+    isFetching,
+    fetchPreviousPage,
+    hasPreviousPage,
+  } = useFetchSessionViewProcessEvents(sessionEntityId, jumpToEvent);
+
+  const hasData = data && data.pages.length > 0 && data.pages[0].events.length > 0;
+  const renderNoData = () => {
+    return (
+      <EuiEmptyPrompt
+        data-test-subj="sessionViewProcessEventsEmpty"
+        title={
+          <h2>
+            <FormattedMessage
+              id="xpack.sessionView.emptyDataMessage"
+              defaultMessage="No data to render"
+            />
+          </h2>
+        }
+        body={<p>No process events found for this query.</p>}
+      />
+    );
+  };
+
+  const renderProcessTree = () => {
```

Comment:
A lot of these `renderX()` functions feel like they should be their own components.

Reply:
Looking at this again, I think having the empty and error states as part of the sessionview component makes sense. If we were to move it into processtree it would only add more complexity to an already complex component.